tests/suites/0_stateless/18_rbac/18_0001_udf_priv.sh

```diff
@@ -0,0 +1,146 @@
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+. "$CURDIR"/../../../shell_env.sh
+
+echo "=== test UDF priv"
+export TEST_USER_PASSWORD="password"
+export TEST_USER_CONNECT="bendsql --user=test-user --password=password --host=${QUERY_MYSQL_HANDLER_HOST} --port ${QUERY_HTTP_HANDLER_PORT}"
+
+echo "drop user if exists 'test-user'" | $BENDSQL_CLIENT_CONNECT
+echo "DROP FUNCTION IF EXISTS f1;" |  $BENDSQL_CLIENT_CONNECT
+echo "DROP FUNCTION IF EXISTS f2;" |  $BENDSQL_CLIENT_CONNECT
+echo "drop table if exists default.t;" | $BENDSQL_CLIENT_CONNECT
+echo "drop table if exists default.t2;" | $BENDSQL_CLIENT_CONNECT
+
+echo "CREATE FUNCTION f1 AS (p) -> (p)" | $BENDSQL_CLIENT_CONNECT
+echo "CREATE FUNCTION f2 AS (p) -> (p)" | $BENDSQL_CLIENT_CONNECT
+echo "create table default.t(i UInt8 not null);" | $BENDSQL_CLIENT_CONNECT
+echo "create table default.t2(i UInt8 not null);" | $BENDSQL_CLIENT_CONNECT
+
+## create user
+echo "create user 'test-user' IDENTIFIED BY '$TEST_USER_PASSWORD'" | $BENDSQL_CLIENT_CONNECT
+echo "grant insert, delete, update, select on default.t to 'test-user';" |  $BENDSQL_CLIENT_CONNECT
+echo "grant select on default.t to 'test-user';" |  $BENDSQL_CLIENT_CONNECT
+echo "grant super on *.* to 'test-user';" |  $BENDSQL_CLIENT_CONNECT
+sleep 2;
+
+# error test need privielge f2
+echo "=== No Privilege on f1, f2 ==="
+echo "select f2(f1(1));" | $TEST_USER_CONNECT
+# bug need error
+echo "select add(1, f2(f1(1)));" | $TEST_USER_CONNECT
```

Comment:
Udf in the query plan into a constant expr or string:

```

select add(1, f2(f1(1)));

plan is Query { s_expr: SExpr { plan: EvalScalar(EvalScalar { items: [ScalarItem { scalar: ConstantExpr(ConstantExpr { span: Some(7..24), value: Number(2_u8) }), index: 1 }] }), children: [SExpr { plan: Scan(Scan { table_index: 0, columns: {0}, push_down_predicates: None, limit: None, order_by: None, prewhere: None, agg_index: None, statistics: Statistics { statistics: None, col_stats: {0: None} } }), children: [], original_group: None, rel_prop: Mutex { data: None, poisoned: false, .. }, stat_info: Mutex { data: None, poisoned: false, .. }, applied_rules: AppliedRules { rules: RuleSet { rules: RoaringBitmap<[]> } } }], original_group: None, rel_prop: Mutex { data: None, poisoned: false, .. }, stat_info: Mutex { data: None, poisoned: false, .. }, applied_rules: AppliedRules { rules: RuleSet { rules: RoaringBitmap<[18]> } } }, metadata: RwLock { data: Metadata { tables: [TableEntry { catalog: "default", database: "system", name: "one", index: 0, .. }], columns: [BaseTableColumn(BaseTableColumn { table_index: 0, column_index: 0, column_name: "dummy", column_position: Some(1), data_type: Number(UInt8), path_indices: None, leaf_index: Some(0), virtual_computed_expr: None }), DerivedColumn(DerivedColumn { column_index: 1, alias: "add(1, f2(f1(1)))", data_type: Number(UInt8) })], lazy_columns: {}, table_row_id_index: {}, agg_indexes: {}, max_column_position: 0 } }, bind_context: BindContext { parent: Some(BindContext { parent: None, columns: [], bound_internal_columns: {}, aggregate_info: AggregateInfo { aggregate_functions: [], aggregate_arguments: [], group_items: [], output_columns: [], aggregate_functions_map: {}, group_items_map: {}, grouping_sets: None }, windows: WindowInfo { window_functions: [], window_functions_map: {} }, lambda_info: LambdaInfo { lambda_arguments: [], lambda_functions: [], lambda_functions_map: {} }, cte_name: None, cte_map_ref: {}, in_grouping: false, view_info: None, srfs: {}, expr_context: Unknown, allow_internal_columns: true, planning_agg_index: false, window_definitions: {} }), columns: [ColumnBinding { database_name: None, table_name: None, column_position: None, table_index: None, column_name: "add(1, f2(f1(1)))", index: 1, data_type: Number(UInt8), visibility: Visible, virtual_computed_expr: None }], bound_internal_columns: {}, aggregate_info: AggregateInfo { aggregate_functions: [], aggregate_arguments: [], group_items: [], output_columns: [], aggregate_functions_map: {}, group_items_map: {}, grouping_sets: None }, windows: WindowInfo { window_functions: [], window_functions_map: {} }, lambda_info: LambdaInfo { lambda_arguments: [], lambda_functions: [], lambda_functions_map: {} }, cte_name: None, cte_map_ref: {}, in_grouping: false, view_info: None, srfs: {}, expr_context: Unknown, allow_internal_columns: true, planning_agg_index: false, window_definitions: {} }, rewrite_kind: None, formatted_ast: None, ignore_result: false }


SELECT i,nth_value(i, f2(f1(2))) OVER (PARTITION BY i) fv FROM t;

plan is Query { s_expr: SExpr { plan: EvalScalar(EvalScalar { items: [ScalarItem { scalar: BoundColumnRef(BoundColumnRef { span: Some(7..8), column: ColumnBinding { database_name: Some("default"), table_name: Some("t"), column_position: Some(1), table_index: Some(0), column_name: "i", index: 0, data_type: Number(UInt8), visibility: Visible, virtual_computed_expr: None } }), index: 0 }, ScalarItem { scalar: BoundColumnRef(BoundColumnRef { span: None, column: ColumnBinding { database_name: None, table_name: None, column_position: None, table_index: None, column_name: "nth_value(i, f2(f1(2))) OVER (PARTITION BY i)", index: 1, data_type: Nullable(Number(UInt8)), visibility: Visible, virtual_computed_expr: None } }), index: 2 }] }), children: [SExpr { plan: Window(Window { span: Some(9..54), index: 1, function: NthValue(NthValueFunction { n: Some(2), arg: BoundColumnRef(BoundColumnRef { span: Some(19..20), column: ColumnBinding { database_name: Some("default"), table_name: Some("t"), column_position: Some(1), table_index: Some(0), column_name: "i", index: 0, data_type: Number(UInt8), visibility: Visible, virtual_computed_expr: None } }), return_type: Nullable(Number(UInt8)) }), arguments: [ScalarItem { scalar: BoundColumnRef(BoundColumnRef { span: Some(19..20), column: ColumnBinding { database_name: Some("default"), table_name: Some("t"), column_position: Some(1), table_index: Some(0), column_name: "i", index: 0, data_type: Number(UInt8), visibility: Visible, virtual_computed_expr: None } }), index: 0 }], partition_by: [ScalarItem { scalar: BoundColumnRef(BoundColumnRef { span: Some(52..53), column: ColumnBinding { database_name: Some("default"), table_name: Some("t"), column_position: Some(1), table_index: Some(0), column_name: "i", index: 0, data_type: Number(UInt8), visibility: Visible, virtual_computed_expr: None } }), index: 0 }], order_by: [], frame: WindowFuncFrame { units: Range, start_bound: Preceding(None), end_bound: Following(None) } }), children: [SExpr { plan: Scan(Scan { table_index: 0, columns: {0}, push_down_predicates: None, limit: None, order_by: None, prewhere: None, agg_index: None, statistics: Statistics { statistics: Some(TableStatistics { num_rows: Some(0), data_size: Some(0), data_size_compressed: Some(0), index_size: Some(0), number_of_blocks: None, number_of_segments: None }), col_stats: {0: None} } }), children: [], original_group: None, rel_prop: Mutex { data: None, poisoned: false, .. }, stat_info: Mutex { data: None, poisoned: false, .. }, applied_rules: AppliedRules { rules: RuleSet { rules: RoaringBitmap<[]> } } }], original_group: None, rel_prop: Mutex { data: None, poisoned: false, .. }, stat_info: Mutex { data: None, poisoned: false, .. }, applied_rules: AppliedRules { rules: RuleSet { rules: RoaringBitmap<[]> } } }], original_group: None, rel_prop: Mutex { data: None, poisoned: false, .. }, stat_info: Mutex { data: None, poisoned: false, .. }, applied_rules: AppliedRules { rules: RuleSet { rules: RoaringBitmap<[18]> } } }, metadata: RwLock { data: Metadata { tables: [TableEntry { catalog: "default", database: "default", name: "t", index: 0, .. }], columns: [BaseTableColumn(BaseTableColumn { table_index: 0, column_index: 0, column_name: "i", column_position: Some(1), data_type: Number(UInt8), path_indices: None, leaf_index: Some(0), virtual_computed_expr: None }), DerivedColumn(DerivedColumn { column_index: 1, alias: "nth_value(i, f2(f1(2))) OVER (PARTITION BY i)", data_type: Nullable(Number(UInt8)) }), DerivedColumn(DerivedColumn { column_index: 2, alias: "fv", data_type: Nullable(Number(UInt8)) })], lazy_columns: {}, table_row_id_index: {}, agg_indexes: {}, max_column_position: 0 } }, bind_context: BindContext { parent: Some(BindContext { parent: None, columns: [], bound_internal_columns: {}, aggregate_info: AggregateInfo { aggregate_functions: [], aggregate_arguments: [], group_items: [], output_columns: [], aggregate_functions_map: {}, group_items_map: {}, grouping_sets: None }, windows: WindowInfo { window_functions: [], window_functions_map: {} }, lambda_info: LambdaInfo { lambda_arguments: [], lambda_functions: [], lambda_functions_map: {} }, cte_name: None, cte_map_ref: {}, in_grouping: false, view_info: None, srfs: {}, expr_context: Unknown, allow_internal_columns: true, planning_agg_index: false, window_definitions: {} }), columns: [ColumnBinding { database_name: Some("default"), table_name: Some("t"), column_position: Some(1), table_index: Some(0), column_name: "i", index: 0, data_type: Number(UInt8), visibility: Visible, virtual_computed_expr: None }, ColumnBinding { database_name: None, table_name: None, column_position: None, table_index: None, column_name: "fv", index: 2, data_type: Nullable(Number(UInt8)), visibility: Visible, virtual_computed_expr: None }], bound_internal_columns: {}, aggregate_info: AggregateInfo { aggregate_functions: [], aggregate_arguments: [], group_items: [], output_columns: [], aggregate_functions_map: {}, group_items_map: {}, grouping_sets: None }, windows: WindowInfo { window_functions: [], window_functions_map: {} }, lambda_info: LambdaInfo { lambda_arguments: [], lambda_functions: [], lambda_functions_map: {} }, cte_name: None, cte_map_ref: {}, in_grouping: false, view_info: None, srfs: {}, expr_context: Unknown, allow_internal_columns: true, planning_agg_index: false, window_definitions: {} }, rewrite_kind: None, formatted_ast: None, ignore_result: false }

insert into t values (f2(f1(1)));

plan is Insert { catalog: "default", database: "default", table: "t", table_id: 294593, schema: TableSchema { fields: [TableField { name: "i", default_expr: None, data_type: Number(UInt8), column_id: 0, computed_expr: None }], metadata: {}, next_column_id: 1 }, overwrite: false }
source is data: "(f2(f1(1)))", start: 21

```

Reply:
cc @andylokandy @sundy-li 

For the insert, `Maybe we can forbidden udf in insert source?`

Constant can not limit in plan.

Reply:
We don't need to check the udf auth if it's already be constantly folded.